### PR TITLE
Devtools/verbose visualization

### PR DESCRIPTION
**Pull request by t-santiagode**
## Description
Adds an extra layer to give fluid handles a bit more clarity potentially necessary for future editing UI. Also makes sharedCounter and sharedString more consistent with how sharedCell displays its information

image

### CODE REVIEW

File: packages/tools/devtools/devtools-view/src/components/data-visualization/FluidHandleView.tsx
Comment (Josmithr) [inline packages/tools/devtools/devtools-view/src/components/data-visualization/FluidHandleView.tsx:92]: I mentioned this in our call, but I think we should hold off on adding the handle-copying until we have the ability to edit values to insert handle IDs. Until then, I think it will just likely confuse users. Waiting will also let us settle on a UI design for the editing buttons and such before adding the copy button.

File: packages/tools/devtools/devtools-view/src/components/data-visualization/FluidHandleView.tsx
Comment (Josmithr) [inline packages/tools/devtools/devtools-view/src/components/data-visualization/FluidHandleView.tsx:105]: This is the only header here, so I think we can just call it "header".

File: packages/tools/devtools/devtools-view/src/components/data-visualization/FluidHandleView.tsx
Comment (Josmithr) [inline packages/tools/devtools/devtools-view/src/components/data-visualization/FluidHandleView.tsx:111]: Nitpick: What do you think about "value" instead of "FluidObject"? Or "data"?

File: packages/tools/devtools/devtools-view/src/components/data-visualization/FluidTreeView.tsx
Comment (Josmithr) [inline packages/tools/devtools/devtools-view/src/components/data-visualization/FluidTreeView.tsx:26]: Nit: Can we revert this line change?

**Review by Josmithr — COMMENTED** [review]
This looks great to me! My only major comment is that I think we should hold off on the handle ID copy button. Everything else looks great! Just a couple of nit-pick comments 😋

We'll want to update the sample screenshot once you've addressed the comments too.

File: packages/tools/devtools/devtools-view/src/components/data-visualization/FluidHandleView.tsx
Comment (t-santiagode) [reply packages/tools/devtools/devtools-view/src/components/data-visualization/FluidHandleView.tsx:105]: If visualTree is undefined than it defines its own header. If I change header2 to just header it compalins about it being "already defined in this scope" since its a const (I think)

File: packages/tools/devtools/devtools-view/src/components/data-visualization/FluidHandleView.tsx
Comment (t-santiagode) [reply packages/tools/devtools/devtools-view/src/components/data-visualization/FluidHandleView.tsx:111]: I think data would go well, it woudld then be consistent with how sharedCell looks with how it shows its values. Will change it

File: packages/tools/devtools/devtools-view/src/components/data-visualization/FluidHandleView.tsx
Comment (t-santiagode) [reply packages/tools/devtools/devtools-view/src/components/data-visualization/FluidHandleView.tsx:105]: If visualTree is undefined than it defines its own header. If I change header2 to just header it compalins about it being "already defined in this scope" since it's a const (I think). What are your thoughts? I imagine some other name makes sense but I'm not sure.

File: packages/tools/devtools/devtools-view/src/components/data-visualization/FluidHandleView.tsx
Comment (t-santiagode) [reply packages/tools/devtools/devtools-view/src/components/data-visualization/FluidHandleView.tsx:92]: I should just remove it rather than commenting it, right? Also another option to keep most of my changes in my branch until its ready to merge completely. Would that make sense to do? Or should I just make these smaller changes anyways?

Comment (Josmithr) [reply packages/tools/devtools/devtools-view/src/components/data-visualization/FluidHandleView.tsx:92]: Yeah, we should remove it altogether. 

As for how to manage your changes - I'm a big advocate for small, scoped PRs. So, I think merging in small changes when we identify that they are ready is the way to go. Merging this PR (with just the tree visuals changes) is a great example IMO.

File: packages/tools/devtools/devtools-view/src/components/data-visualization/FluidHandleView.tsx
Comment (Josmithr) [reply packages/tools/devtools/devtools-view/src/components/data-visualization/FluidHandleView.tsx:105]: Oh, I completely missed that. That might be a good motivation to wrap the remainder of the method in an `else` block (so we can use the same variable name).

Comment (t-santiagode) [reply packages/tools/devtools/devtools-view/src/components/data-visualization/FluidHandleView.tsx]: I think this is resolved.

File: packages/tools/devtools/devtools-view/src/components/data-visualization/FluidTreeView.tsx
Comment (t-santiagode) [reply packages/tools/devtools/devtools-view/src/components/data-visualization/FluidTreeView.tsx]: Reverted the change.

**Review by alexvy86 — APPROVED** [review]
Looks alright to me. @jikim-msft I'd like your thoughts as well, since you're more familiar with the visualization part.

**Comment by jikim-msft** [comment]
> Looks alright to me. @jikim-msft I'd like your thoughts as well, since you're more familiar with the visualization part.

I will wait until @t-santiagode pushes more changes in and start the review 😄 . Thank you for the heads up!

**Comment by Josmithr** [comment]
@t-santiagode @jikim-msft This PR looks like it's good to merge. Should we do so? Or are we waiting on something else to go in first?

**Review by Josmithr — APPROVED** [review]

**Pull request closed without merging**
